packages/data/src/components/with-dispatch/index.js

```diff
@@ -44,7 +44,7 @@ const withDispatch = ( mapDispatchToProps ) => createHigherOrderComponent(
 
 				proxyDispatch( propName, ...args ) {
 					// Original dispatcher is a pre-bound (dispatching) action creator.
-					mapDispatchToProps( this.props.registry.dispatch, this.props.ownProps )[ propName ]( ...args );
+					mapDispatchToProps( this.props.registry.dispatch, this.props.ownProps, this.props.registry.select )[ propName ]( ...args );
```

Comment:
Just a thought, what if the third argument is the registry instead of the select function? I think it's slightly better because it can be made consistent with `withSelect`?

Reply:
There are only 2 params in `withSelect` passed to `mapSelectToProps`:

https://github.com/WordPress/gutenberg/blob/fba803e46ba930b09e3763d224e5abfdf0275ed2/packages/data/src/components/with-select/index.js#L42

Can you further explain your thought? I'm open to change the implementation. I'm not that much concerned what we pass as 3rd param. The registry might make sense, too.

Reply:
The idea is that we could (I don't have a use case now) add a registry argument for withSelect as well which will be inline with withDispatch.

Reply:
It's an interesting way to more directly solve the root problem, which is that we don't know for which registry on which to select data (otherwise we could use `wp.data.select`). It could possibly also resolve future unforeseen requirements.

I think it feels a bit less ergonomic. In some ways I like that - it makes it feel even less natural to select from `withDispatch`. In-place destructuring could make it less painful to work with, but can also lose some clarity as to what the arguments actually are. For an advanced use-case, that might be okay.

Reply:
Does it mean that we agree to pass `this.props.registry` as 3rd param instead?

Reply:
I think it's the better option, yes.